aws-lumberyard-dev/o3de-multiplayersample
# Add teleport component

Adds a basic teleport component which detects a collision and moves the colliding player to a new location.

Notes on why its **draft**:
 * It's not networked (yet?) as I wasn't sure it needed to be, give its attached to stationary entities (but maybe that won't always be the case?)
 * After teleporting, the player seems to lack forward walking animation (but retains side walking, jumping, and fire projectile animation)
 * It lacks tests.

### Testing

1. add new _Network Teleport Component_ to an entity with a PhysX Collider (trigger mode)
2. create and/or select an entity to act as the teleport destination
![image](https://user-images.githubusercontent.com/34254888/179121181-96fe045a-ad48-49e2-b9c0-649dab91ed27.png)
3. hit `ctrl + G` and navigate the player into the entity with the Teleport component, the player should be transported to the "Destination" entity.
4. See "Teleporter" logs of event:
![image](https://user-images.githubusercontent.com/34254888/179121317-ba760510-d010-4187-ac1e-ac68cc0c46ea.png)
